game/scripts/vscripts/abilities/swiper/boss_swiper_reapers_rush.lua

```diff
@@ -0,0 +1,225 @@
+		local direction = (target - caster:GetAbsOrigin()):Normalized()
+
+		DebugDrawBoxDirection(caster:GetAbsOrigin(), Vector(0,-width / 2,0), Vector(distance,width / 2,50), direction, Vector(255,0,0), 1, castTime)
+		DebugDrawCircle(target + Vector(0,0,32), Vector(255,0,0), 128, width, false, castTime + 2.0)
+	end
+	return true
+end
+
+--------------------------------------------------------------------------------
+
+function boss_swiper_reapers_rush:GetPlaybackRateOverride()
+	return 0.275
+end
+
+--------------------------------------------------------------------------------
+
+function boss_swiper_reapers_rush:OnSpellStart()
+	if IsServer() then
+		local caster = self:GetCaster()
+		local target = self:GetCursorPosition()
+		local distance = (target - caster:GetAbsOrigin()):Length2D()
+		local direction = (target - caster:GetAbsOrigin()):Normalized()
+
+		caster:Stop()
+
+		local modifier = caster:AddNewModifier(caster, self, "boss_swiper_reapers_rush_active", {})
+		local modifierTable = {}
+		modifierTable.speed = self:GetSpecialValueFor("speed")
+		modifierTable.distance = distance
+		modifier:InitRush(modifierTable)
+	end
+end
+
+------------------------------------------------------------------------------------
+
+boss_swiper_reapers_rush_active = class(ModifierBaseClass)
+
+------------------------------------------------------------------------------------
+
+function boss_swiper_reapers_rush_active:IsPurgable()
+	return false
+end
+
+------------------------------------------------------------------------------------
+
+function boss_swiper_reapers_rush_active:GetActivityTranslationModifiers()
+    return "haste"
+end
+
+--------------------------------------------------------------------------------
+
+function boss_swiper_reapers_rush_active:GetOverrideAnimationRate()
+	return 2.0
+end
+
+------------------------------------------------------------------------------------
+
+function boss_swiper_reapers_rush_active:GetOverrideAnimation()
+	return ACT_DOTA_RUN
+end
+
+------------------------------------------------------------------------------------
+
+function boss_swiper_reapers_rush_active:GetOverrideAnimationWeight(params)
+    return 1.0
+end
+
+------------------------------------------------------------------------------------
+
+function boss_swiper_reapers_rush_active:DeclareFunctions()
+	local funcs = {
+        MODIFIER_PROPERTY_OVERRIDE_ANIMATION,
+        MODIFIER_PROPERTY_OVERRIDE_ANIMATION_WEIGHT,
+        MODIFIER_PROPERTY_OVERRIDE_ANIMATION_RATE,
+		MODIFIER_PROPERTY_TRANSLATE_ACTIVITY_MODIFIERS
+	}
+
+	return funcs
+end
+
+------------------------------------------------------------------------------------
+
+function boss_swiper_reapers_rush_active:CheckState()
+    local state = {
+        [MODIFIER_STATE_COMMAND_RESTRICTED] = true,
+        [MODIFIER_STATE_NO_UNIT_COLLISION] = true,
+        [MODIFIER_STATE_FLYING_FOR_PATHING_PURPOSES_ONLY] = true,
+        [MODIFIER_STATE_STUNNED] = true,
```

Comment:
more a question then a comment, why it has no collision and is stunned?

Reply:
It's a good practice to disable units during such movement skills. Nothing should interrupt it like autoattack or bodyblocking.